src/mito2/src/memtable/bulk/part_reader.rs

```diff
@@ -152,8 +176,14 @@ impl BulkPartRecordBatchIter {
         // Apply projection first.
         let projected_batch = self.apply_projection(record_batch)?;
         // Apply combined filtering (both predicate and sequence filters)
+        // For BulkPartRecordBatchIter, we don't have row group information.
+        let skip_fields = match self.context.pre_filter_mode() {
+            PreFilterMode::All => false,
+            PreFilterMode::SkipFields => true,
+            PreFilterMode::SkipFieldsOnDelete => true,
```

Comment:
Unable to check if the batch includes any delete operations?

Reply:
It's possible but requires iterating the op types. So I prefer to keep things simple here. Also, this PR is already large enough.

Reply:
Got it